### ISSUE TITLE
Can't get Client to work... Auth Files on Linux

### ISSUE THREAD

**Issue by Snify89** [The-Cataclysm-Preservation-Project/TrinityCore#382]
Since there is only 1 documentation available here:
https://github.com/The-Cataclysm-Preservation-Project/TrinityCore/wiki/Setup-and-troubleshooting-for-connecting

I managed to get through the compilation, etc. and my server is running.

However I can't make the client work so far...

The steps I took was as follows:

1. Executed ClientDownloader and WoWRegenaration to get the full client. (on Windows)
2. I extracted the client data (mmaps, vmaps , etc...) (on Linux)
3. Compiled and started the server (on Linux) (SSL 1.1.1f)
4. Database (MySQL 8), Authserver (bnetserver), worldserver runs fine and it applied all hotfix and world db entries.
5. I applied connection_patcher with the full client (on Linux)
with the following log:

```
Creating patched binary...
Win32 client...
Patching client binary...
patching redirect RSA Modulus
Found offset 7961504
Patching done.
Patching battle.net binary...
patching Portal
Found offset 1694224
patching Bnet
Found offset 864857
patching Signature
Found offset 134009
Patching done.
Base module doesn't exist, downloading it...
EX: Could not resolve server address for auth module download
An error occurred. Press ENTER to continue...
```
6. copied patched files (dll and exe) to my full client on Windows
7. replaced dll and exe files properly
8. changed realmlist IP address to my local 192.x IP on database
9. changed config.wtf with 
SET portal "192.168.2.176"
SET realmlist "192.168.2.176"
10. Started client and tried to login with my created account "test@test" 
11. It says, that the version cannot be verified.
12. Deleted launcher.exe from full client
13. Now I'm stuck and I can't login with the error : "Could not establish connection"

So I probably have 2 things missing:
1. Server binding won't let me connect to the IP
2. The thing with the auth files...

The auth files are a mystery to me.
How can I make this work with tools and server on a Linux machine?

Any help is appreciated thank you :)

**Comment by funjoker** [comment]
Wine emulates the same paths like in windows. 
Move the auth files to the correct folder. If you patched it on windows you likely need to move the patched auth file as well. Best copy all auth. files files

**Comment by Snify89** [comment]
> Wine emulates the same paths like in windows. Move the auth files to the correct folder. If you patched it on windows you likely need to move the patched auth file as well. Best copy all auth. files files

Thank you for your reply...
I don't use Wine at all.
Since I compiled all tools (including connection_patcher) and server in Linux, Wine won't work here - until I compile it for Windows?!

I can't test/execute the connection_patcher on windows.

How would this work on/with Wine?

I quite don't get the thing at all with the auth files...
Could you please elaborate this exact procedure a bit further?
Are those auth files needed by the server and/or client (only)?
Do I need existing (original/legit) auth files and they will be patched?
Are those auth files universal or is this needed for each client/server (ID?)

Bottom line: 
I have server and tools/connection_patcher only available on Linux and the client on Windows
How can I make this work?

Thank you very much again for your help

**Comment by Ovahlord** [comment]
The client needs these modules. The patcher failed because blizzard has taken them offline so you gotta go with our backup repository

**Comment by Snify89** [comment]
> The client needs these modules. The patcher failed because blizzard has taken them offline so you gotta go with our backup repository

Ok thank you.
I did the following procedure now on my linux machine ( I DO/DID NOT USE WINE! ):

1. I created a path in my home directory (/root/.wine/drive_c/ProgramData/Blizzard Entertainment/Battle.net/Cache)
2. I copied all the files from https://github.com/The-Cataclysm-Preservation-Project/AuthModules/tree/master/Windows/ProgramData/Blizzard%20Entertainment/Battle.net/Cache into my local linux directory
3. I ran connection_patcher again (e.g. "./connection_patcher /path/to/my/wow/client/wow-64.exe")

Following log appears

```
TrinityCore rev. 0d3dfa719e56 2022-11-13 12:14:01 +0100 (master branch) (Unix, RelWithDebInfo, Static) (connection_patcher)
<Ctrl-C> to stop.

Creating patched binary...
Win64 client...
Patching client binary...
patching redirect RSA Modulus
Found offset 9924352
Patching done.
Patching battle.net binary...
patching Portal
Found offset 2707952
patching Bnet
Found offset 1288780
patching Signature
Found offset 180238
Patching done.
Patching module...
patching Password
Found offset 5066
Patching module finished.
Successfully created your patched binaries.
```

I copied the contents from 
/root/.wine/drive_c/ProgramData/Blizzard Entertainment/Battle.net/Cache
 to my windows machine (C:\ProgramData\Blizzard...)

I run the patched .exe and still nothing.

No files in 
/root/.wine/drive_c/ProgramData/Blizzard Entertainment/Battle.net/Cache
have changed as it seems...

What are the x64 and x86 folders in the AuthModules do, and are they needed?!
Do I need to copy them somewhere?

Thank you so much clearing things up.
